app/ica_module.go

```diff
@@ -0,0 +1,51 @@
+package app
+
+import (
+	"encoding/json"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	ica "github.com/cosmos/ibc-go/v6/modules/apps/27-interchain-accounts"
+	icagenesistypes "github.com/cosmos/ibc-go/v6/modules/apps/27-interchain-accounts/genesis/types"
+	icahostkeeper "github.com/cosmos/ibc-go/v6/modules/apps/27-interchain-accounts/host/keeper"
+)
+
+const (
+	arabicaChainID = "arabica-11"
+	mochaChainID   = "mocha-4"
+)
+
+// IcaModule implements the AppModule interface.
+var _ module.AppModule = IcaModule{}
+
+// IcaModule is a wrapper around the ICA module that allows for a custom DefaultGenesis function.
+type IcaModule struct {
+	ica.AppModule
+}
+
+// NewIcaModule creates a new ICA module with a custom DefaultGenesis function.
+func NewIcaModule(chainID string, icaHostKeeper icahostkeeper.Keeper) module.AppModule {
+	if chainID == arabicaChainID || chainID == mochaChainID {
+		// These testnets went through the v2 activation height on a
+		// celestia-app release that did not use a custom DefaultGenesis
+		// function.
+		return ica.NewAppModule(nil, &icaHostKeeper)
+	}
```

Comment:
Does this matter now? Will arabica and mocha call `DefaultGenesis` again?

Reply:
Arabica and Mocha nodes that sync from genesis will execute all blocks including the v2 activation height at which point I expect them to invoke `DefaultGenesis`. If we want to avoid an app hash mismatch for them, I think we need this.